protocols/floodsub/src/lib.rs

```diff
@@ -635,7 +635,8 @@ fn handle_packet_received(
             let subscribed_topics = inner.subscribed_topics.read();
             topics
                 .iter()
-                .any(|t| subscribed_topics.iter().any(|topic| topic.hash() == t))
+                .any(|t| subscribed_topics.iter().any(|topic| 
+                    topic.descriptor.clone().get_name().to_string() == *t))
```

Comment:
does allocation need to be done here? can't we check equality without reallocating?

Reply:
I'll look into that; although I'm not sure how it might be done off the top of my head.